dev_scripts_helpers/documentation/render_images.py

```diff
@@ -84,128 +84,10 @@ def _get_rendered_file_paths(
-    the entry corresponding to that hash.
-    """
-
-    def __init__(self, cache_file: str):
-        """
-        Initialize the ImageHashCache.
-
-        :param cache_file: Path to the cache file
-        """
-        hdbg.dassert_isinstance(cache_file, str)
-        hdbg.dassert_ne(cache_file, "")
-        self.cache_file = cache_file
-        # Load the cache from the file, if it exists, or start with an empty
-        # cache.
-        if os.path.exists(self.cache_file):
-            _LOG.debug("Loading cache from %s", self.cache_file)
-            self.cache = self._load()
-        else:
-            _LOG.debug("No cache file found at %s", self.cache_file)
-            self.cache = {}
-
-    def __contains__(self, entry_key: str) -> bool:
-        """
-        Check if an entry is in the cache.
-        """
-        return entry_key in self.cache
-
-    def compute_hash(
-        self, image_code: str, image_code_type: str, out_file: str
-    ) -> Tuple[str, dict]:
-        """
-        Compute a hash of the needed computation inputs.
-
-        :param image_code: The code of the image
-        :param image_code_type: Type of the image code (e.g.,
-            "plantuml", "mermaid")
-        :param out_file: Path to the output file where the image will be
-            saved
-        :return: The hash of the inputs and the inputs
-        """
-        # Compute the hash of the image code.
-        image_code_hash = hashlib.sha256(image_code.encode()).hexdigest()
-        # Create a dictionary with the inputs
-        cache_value = {
-            "image_code_hash": image_code_hash,
-            "image_code_type": image_code_type,
-            "out_file": out_file,
-        }
-        # Compute hash of the entry.
-        cache_key = out_file
-        return cache_key, cache_value
-
-    def update_cache(
-        self, cache_key: str, cache_value: str, *, assert_no_key: bool = False
-    ) -> bool:
-        """
-        Update the cache with a new entry.
-        """
-        if assert_no_key:
-            hdbg.dassert_not_in(cache_key, self.cache)
-        cache_updated = (
-            cache_key not in self.cache or self.cache[cache_key] != cache_value
-        )
-        if cache_updated:
-            self.cache[cache_key] = cache_value
-            # Save the cache to the file.
-            self._save()
-        return cache_updated
-
-    def _load(self) -> dict:
-        """
-        Load the hash cache from a file.
-
-        :return: Dictionary mapping image hashes to rendered file paths
-        """
-        hdbg.dassert_file_exists(self.cache_file)
-        return hio.from_json(self.cache_file)
-
-    def _save(self) -> None:
-        """
-        Save the hash cache to a file.
-        """
-        hio.to_json(self.cache_file, self.cache)
-
-
 # #############################################################################
 
 
+@hcacsimp.simple_cache(write_through=True)
```

Comment:
Can we explain in a comment why we set write_through here?

If we are overwriting a default param, we should explain it for the principle of minimum surprise